### PR TITLE
feat: Initiate loadGeneratorInterface

### PR DESCRIPTION
**Pull request by kushthedude**
Refers #211

### CODE REVIEW

**Comment by leecalcote** [comment]
https://github.com/layer5io/meshery/pull/1156#discussion_r443153953

**Comment by kushthedude** [comment]
> #1156 (comment)

No it should not change, the only change will happen due to the SupportedLoadTestMethods which we will invoke through SMPS.

File: models/load-test.go [hunk 1]
```diff
@@ -28,6 +28,20 @@ func (l LoadGenerator) Name() string {
 	return string(l)
 }
 
+// SupportedLoadTestMethods - represents the load test method type
+type SupportedLoadTestMethods int
```
Comment (leecalcote) [inline models/load-test.go:32]: What factors weigh on a decision to or to not use the SMPS traffic.proto? Would that be a good or bad thing?

**Comment by leecalcote** [comment]
@kushthedude this works for me in fortio tests. There are a couple merge conflicts to address in load_test_handler.go.

**Comment by kushthedude** [comment]
@leecalcote conflicts resolved

**Comment by leecalcote** [comment]
@kushthedude I wasn't able to test out the wrk2 (build in docker), but I assume that we're good to go with this first portion of the interface.

**Comment by kushthedude** [comment]
> @kushthedude I wasn't able to test out the wrk2 (build in docker), but I assume that we're good to go with this first portion of the interface.

It is working too, I had tested it before changes.

**Pull request merged**